packages/snap-platforms/common/README.md

```diff
@@ -2,11 +2,101 @@
+## SnapTemplates Usage
+To use a plugin in [SnapTemplates](https://github.com/searchspring/snap/blob/main/docs/TEMPLATES_ABOUT.md), it can be defined and configured in the `config.plugins` section.
+
+```jsx
+plugins: {
+	common: {
+		scrollToTop: {
+			enabled: true
+		}
+	}
+}
+```
+
+## backgroundFilters
+Allows you to set up background filters. You can configure filters for tags, collections, or other fields.
+
+> [!NOTE]
+> The common backgroundFilters plugin provides a generic manual way of setting background filters; however, when `shopify`, `bigcommerce` or `magento2` are defined in the `config.plugins`, additional plugins are attached to handle platform specific functionality, for example, setting `backgroundFilters`.
+
+| Configuration Option | Description | Type | Required |
+|----------------------|-------------|------|---------|
+| `plugins.common.backgroundFilters` | Background filter configurations | Object | ➖ |
+| `plugins.common.backgroundFilters.filters[]` | Background filter definitions | Array | ➖ |
+| `plugins.common.backgroundFilters.filters[].type` | Defines if filter should be 'value' or 'range' type | 'value' \| 'range' | ✔️ |
+| `plugins.common.backgroundFilters.filters[].field` | Defines filter field name | string | ✔️ |
+| `plugins.common.backgroundFilters.filters[].value` | Defines filter value. If `type` is 'value', this must be a string, otherwise if `type` is 'range', this must be an object with `low` and `high` properties | string \| { low: number, high: number } | ✔️ |
+| `plugins.common.backgroundFilters.filters[].controllerIds` | Defines which controllers the filter should apply to | (string \| regexp)[]  | ➖ |
+| `plugins.common.backgroundFilters.filters[].controllerType` | Defines which controller types the filter should apply to | (string)[] | ➖ |
+
+```jsx
+plugins: {
+	common: {
+		backgroundFilters: {
+			filters: [{
+				type: 'value',
+				field: 'ss_tags',
+				value: 'instock'
+			},
+			{
+				type: 'value',
+				field: 'collection',
+				value: 'mens'
+			},
+			{
+				type: 'value',
+				field: 'custom',
+				value: '1',
+			},
+			{
+				type: 'range',
+				field: 'price',
+				value: { low: 10, high: 20 },
+			}],
+		}
+	}
+}
+```
+
+
+## scrollToTop
+Configures the behavior of scrolling to the top of the page after a search has occurred.
+
+> [!NOTE]
+> This plugin only applies to search and category pages (search controllers)
+
+| Configuration Option | Description | Type | Default |
+|----------------------|-------------|------|---------|
+| `plugins.common.scrollToTop` | Scroll to top plugin configuration | Object | ➖ |
+| `plugins.common.scrollToTop.enabled` | Enables plugin | boolean | true |
+| `plugins.common.scrollToTop.selector` | Query selector to scroll to | string | 'body' |
+| `plugins.common.scrollToTop.options` | [`window.scroll` options configuration](https://developer.mozilla.org/en-US/docs/Web/API/Window/scroll#options) | Object | `{ top: 0, left: 0, behavior: 'smooth' }` |
```

Comment:
Maybe make more generic so that documentation applies to Snap and Snap Templates.